test/form_helper_test.rb

```diff
@@ -1188,4 +1188,56 @@
       end
     end
   end
+
+  describe "#datalist" do
+    let(:values) { Hash['it' => 'Italy', 'us' => 'United States'] }
+
+    it "renders" do
+      actual = view.form_for(:book, action) do
+        datalist :store, values, 'books'
+      end.to_s
+
+      actual.must_include %(<input type="text" name="book[store]" id="book-store" list="books">\n<datalist id="books">\n<option value="Italy">it</option>\n<option value="United States">us</option>\n</datalist>)
```

Comment:
If we do that the options won't display right. You can see at [here](https://jsbin.com/yutufobizu/edit?html,output)